ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/resources/KsqlResourceTest.java

```
@@ -791,23 +799,41 @@ public void shouldDistributeTerminateQuery() {
   }
 
   @Test
-  public void shouldDistributeTerminateQueryWithoutValidation() {
-    // Why? Because currently if the server receives a single request containing two statements:
-    // `CREATE STREAM FOO AS blah;`
-    // `TERMINATE csas_foo_0;`
-    // Then its possible that the terminate line is valid, in that it will terminate the query
-    // started by the first line, but the server can no validate this as the CSAS may not have
-    // be actioned by the StatementExecutor yet.
-
-    // Given:
-    final String terminateSql = "TERMINATE some_id;";
+  public void shouldThrowOnTerminateUnknownQuery() {
```

We can now fail validation when the user tries to TERMINATE a unknown query!